src/history/hs_rec.c

```diff
@@ -831,17 +838,21 @@ __hs_delete_key_from_ts_int(
  */
 int
 __wt_hs_delete_key_from_ts(
-  WT_SESSION_IMPL *session, uint32_t btree_id, const WT_ITEM *key, wt_timestamp_t ts, bool reinsert)
+  WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key, wt_timestamp_t ts, bool reinsert)
 {
     WT_DECL_RET;
+    WT_HS_KEY hs_key;
```

Comment:
Whats the rationale for not using WT_HS_KEY here?

Reply:
This is a __wt function. If I want to use WT_HS_KEY here, I need to define WT_HS_KEY outside of the hs module. I think it is better to make this structure internal to the hs module for now.

Reply:
I think moving it outside and into `wt_internal.h` makes more sense if we're going to create these objects. All callers to `__wt_hs_blah` would be able to use them.

Reply:
I don't think that is a good idea because we are not including everything in the HS key to the WT_HS_KEY, which may be confusing. I would suggest revisit that after we have done the restructure project.

Reply:
I think not including everything that makes up the history store key in `WT_HS_KEY` is confusing. But I'm unsure if that would be fixed by restructure history store project.

Reply:
In the first stage of the restructure, no. In the second stage, we will try to remove the counter from the user interface and make it internal to the history store cursor. However, still we don't include the start_ts in the key.